### PR TITLE
MySQL Design and Data Access Model

### PR DESCRIPTION
**Pull request by Selinaqianli**

### CODE REVIEW

File: src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/admin.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE admin (
+                      id BIGINT AUTO_INCREMENT PRIMARY KEY,
+                      username VARCHAR(255) NOT NULL UNIQUE,
+                      email VARCHAR(255) NOT NULL UNIQUE,
+                      password VARCHAR(255) NOT NULL,
+                      machine_id BIGINT,
```
Comment (Yiwen0612) [inline src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/admin.sql]: the latest design is: after admin login, system will provide a drop down list of machine id for them, one (admin)to many(machine) , this need to be changed, need discuss it first

File: src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/allergy.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE allergy (
+                         id BIGINT AUTO_INCREMENT PRIMARY KEY,
+                         name VARCHAR(255) NOT NULL,
```
Comment (Yiwen0612) [inline src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/allergy.sql]: my original design is: when admin set the drink in our system (match the ingredient name, amount in a correct position step), they will set this drinkès allergy info, and when user order a drink with this ingredient, the allergy info will show up

File: src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/drink.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+                       name VARCHAR(255) NOT NULL,
+                       categoryId BIGINT NOT NULL,
+                       imageURL VARCHAR(255),
+                       price DOUBLE NOT NULL,
+                       priceCurrency VARCHAR(255) NOT NULL,
+                       description VARCHAR(255),
+                       isActive BOOLEAN DEFAULT true,
+                       isCustomized BOOLEAN DEFAULT false,
+                       userId BIGINT,
+                       machineId BIGINT,
```
Comment (Yiwen0612) [inline src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/drink.sql]: ref

File: src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/drink.sql [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+                       id BIGINT AUTO_INCREMENT PRIMARY KEY,
+                       name VARCHAR(255) NOT NULL,
+                       categoryId BIGINT NOT NULL,
+                       imageURL VARCHAR(255),
+                       price DOUBLE NOT NULL,
+                       priceCurrency VARCHAR(255) NOT NULL,
+                       description VARCHAR(255),
+                       isActive BOOLEAN DEFAULT true,
+                       isCustomized BOOLEAN DEFAULT false,
+                       userId BIGINT,
```
Comment (Yiwen0612) [inline src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/drink.sql]: if it is customize, user id is user id; if it is not customized, user id means admin id, just let u know, no changes need here

File: src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/drinkcategory.sql [hunk 5]
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE drinkcategory (
```
Comment (Yiwen0612) [inline src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/drinkcategory.sql]: if we only have alcohol and non-alcohol, maybe this table can be enumÉ same as role tableÉ

File: src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/ingredient.sql [hunk 6]
```diff
@@ -0,0 +1,11 @@
+CREATE TABLE ingredient (
+                            id BIGINT AUTO_INCREMENT PRIMARY KEY,
+                            name VARCHAR(255) NOT NULL,
+                            machineId BIGINT NOT NULL,
```
Comment (Yiwen0612) [inline src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/ingredient.sql]: ref here?

File: src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/user.sql [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+                      password VARCHAR(255) NOT NULL,
+                      isEnabled BOOLEAN DEFAULT true,
+                      dateOfBirth DATE,
+                      imageURL VARCHAR(255),
+                      lastSobrietyTestScore DOUBLE,
+                      lastSobrietyTestAt TIMESTAMP,
+                      signedInAt TIMESTAMP,
+                      createdAt TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+                      modifiedAt TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
```
Comment (Yiwen0612) [inline src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/user.sql:20]: should we have a field for user to set their allery?

File: src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/allergy.sql (hunk 2 above)
Comment (Yiwen0612) [reply src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/allergy.sql]: uhmmmm i am thinking... it s a enum or database table?

Comment (Selinaqianli) [reply src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/allergy.sql]: I can be a table, an ingredient may be associated to many allergy. It's not ideal to store a list of items in one ingredient column

File: src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/drink.sql (hunk 4 above)
Comment (Selinaqianli) [reply src/main/java/ca/uwaterloo/drinkmasterapi/sql/creation/drink.sql]: added comments here

**Review by Yiwen0612 — APPROVED** [review]

**Pull request merged**